src/routes/transactions/mappers/common/native-staking.mapper.ts

```diff
@@ -168,6 +179,66 @@ export class NativeStakingMapper {
     });
   }
 
+  /**
+   * Maps the {@link NativeStakingWithdrawTransactionInfo} for the given
+   * native staking deployment's `withdraw` call
+   *
+   * @param args.chainId - the chain ID of the native staking deployment
+   * @param args.to - the address of the native staking deployment
+   * @param args.value - the value of the withdraw transaction
+   * @param args.transaction - the transaction object for the withdraw
+   * @returns {@link NativeStakingWithdrawTransactionInfo} for the given native staking deployment
+   */
+  // TODO: refactor common logic between mapDepositInfo and mapValidatorsExitInfo and mapWithdrawInfo
+  public async mapWithdrawInfo(args: {
+    chainId: string;
+    to: `0x${string}`;
+    value: string | null;
+    transaction: MultisigTransaction | ModuleTransaction | null;
+  }): Promise<NativeStakingWithdrawTransactionInfo> {
+    const [chain, deployment] = await Promise.all([
+      this.chainsRepository.getChain(args.chainId),
+      this.stakingRepository.getDeployment({
+        chainId: args.chainId,
+        address: args.to,
+      }),
+    ]);
+
+    if (
+      deployment.product_type !== 'dedicated' ||
+      deployment.chain === 'unknown' ||
+      deployment.status === 'unknown'
+    ) {
+      throw new NotFoundException('Native staking deployment not found');
+    }
+
+    let value = Number(args.value ?? 0);
+
+    if (args.transaction) {
+      const validatorsPublicKeys = args.transaction?.dataDecoded
+        ? this.getPublicKeysFromDataDecoded(args.transaction.dataDecoded)
+        : [];
+
+      const stakes = await this.stakingRepository.getStakes({
+        chainId: args.chainId,
+        validatorsPublicKeys,
+      });
+      value = stakes.reduce((acc, stake) => acc + Number(stake.rewards), 0);
```

Comment:
This disregards the current value of `value`. Does this matter?

Reply:
Yes, the function argument `value` is only considered when no transaction is present (i.e.: confirmation view). Otherwise (i.e.: transaction mapping) the `value` is requested to the `StakingApi`.